src/main/java/org/openrewrite/java/migrate/lombok/LombokValueToRecord.java

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.lombok;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.AnnotationMatcher;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.RemoveAnnotationVisitor;
+import org.openrewrite.java.search.UsesJavaVersion;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Statement;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class LombokValueToRecord extends ScanningRecipe<Map<String, Set<String>>> {
+
+    private static final AnnotationMatcher LOMBOK_VALUE_MATCHER = new AnnotationMatcher("@lombok.Value");
+
+    @Option(displayName = "useExactToString",
+            description = "When set the `toString` format from Lombok is used in the migrated record.")
```

Comment:
from my perspective this would be fine to make the option non-required but personally I would prefer to keep a primitive boolean that is initialized from the default constructor to false so basically like:
```
@Option(displayName = "Add a `toString()` implementation matching Lombok",
            description = "When set the `toString` format from Lombok is used in the migrated record.",
            required = false)
boolean useExactToString;

public LombokValueToRecord() {
        this(false);
}

public LombokValueToRecord(boolean useExactToString) {
        this.useExactToString = useExactToString;
}
```

@knutwannheden @timtebeek  what do you think?

Reply:
There's some serialization that factors in here as well, where we can't be certain the default constructor is called. In other places having a Boolean field then allows us to differentiate, and perhaps using that uniformly would be best then here as well.

Reply:
While it may seem a bit odd, I also think it is more important that we stay consistent with respect to other recipes. In the future we may be able to find a better solution for this. But as Tim says, it has to do with the serialization in the SaaS platform.